contracts/utils/Time.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+/**
+ * @dev Tooling for timepoints, timers and delays
+ */
+library Time {
+    enum ReservedTimestamps {
+        Unset,  // 0
+        Locked, // 1
+        length
+    }
+
+    struct Timer {
+        uint256 _deadline;
+    }
+
+    function getDeadline(Timer memory timer) internal pure returns (uint256) {
+        return timer._deadline;
+    }
+
+    function setDeadline(Timer storage timer, uint256 timestamp) internal {
+        timer._deadline = timestamp;
+    }
+
+    function reset(Timer storage timer) internal {
+        setDeadline(timer, uint256(ReservedTimestamps.Unset));
+    }
+
+    function lock(Timer storage timer) internal {
```

Comment:
The difference between lock and reset is that lock identifies the special without the need for an additional memory slot.

We could use 2 boolean slots, for executed and canceled ... but I think this would be more expensive ... we could also add 2 different states in Timer, like I shown in this issue: https://github.com/ethereum/solidity/issues/11469#issuecomment-853746865